core/livepeernode.go

```diff
@@ -125,3 +125,14 @@ func (n *LivepeerNode) StopEthServices() error {
 
 	return nil
 }
+
+func (n *LivepeerNode) GetServiceURI() *url.URL {
+	return n.serviceURI
+}
+
+func (n *LivepeerNode) SetServiceURI(newUrl *url.URL) {
+	if newUrl != nil && n.serviceURI != nil {
+		*n.serviceURI = *newUrl
+	}
+	n.serviceURI = newUrl //for nil cases
```

Comment:
this should be in `else` branch - right now you're overwriting pointer and it breaks on second call to `SetServiceURI`.
I've added test case in comments.

Reply:
> Part of why this reads a bit strangely is because we're both 1) setting a possibly-nil pointer, and 2) setting the contents of that pointer. Both are done under different conditions.

3) it is always overwriting pointer. So we are not achieving goal of `so it'll always load the latest ServiceURI value`- because on second invocation of `SetServiceURI` we're not setting content of `baseURI` - `LivepeerNode` doesn't have that pointer anymore. 
And we can't just simply set `serviceURI` to be value instead of pointer - `MemoryDriver` still need a way to check if it's empty.

Probably ideal solution here will be 
```golang
	if newUrl != nil && n.serviceURI != nil {
		*n.serviceURI = *newUrl
	} else if n.serviceURI == nil && newUrl != nil {
		n.serviceURI = &url.URL{}
		*n.serviceURI = *newUrl
	}
```

Reply:
> And we can't just simply set serviceURI to be value instead of pointer - MemoryDriver still need a way to check if it's empty.

The MemoryDriver only needs to check that the ServiceURI is non-nil here (which this PR does in `getAbsoluteURI`). Calling `.String()` on an empty URL yields the empty string which is fine.

Storing a value rather than a pointer should still work, unless I'm missing something else.

Reply:
If `.String()` returns empty string for empty `URL` structure then storing value is a way to go